### ISSUE TITLE
this.backdrop.remove is not a function - The tour isn't properly built.

### ISSUE THREAD

**Issue by cenobyte321** [sorich87/bootstrap-tour#507]
I have the bootstrap tour working in an app with AngularJS and Designmodo's startup framework. The problem is that the app crashes randomly when the window width size is below 992px, a navbar-collapse menu is shown and I use the step.goTo function. This happens in Google Chrome. 

The crash error is the following:

```
TypeError: this.backdrop.remove is not a function
    at Tour._hideBackground (http://localhost:8080/resources/bower_components/bootstrap-tour/build/js/bootstrap-tour.js:726:23)
    at Tour._hideBackdrop (http://localhost:8080/resources/bower_components/bootstrap-tour/build/js/bootstrap-tour.js:720:19)
    at Tour.<anonymous> (http://localhost:8080/resources/bower_components/bootstrap-tour/build/js/bootstrap-tour.js:264:19)
    at Tour._callOnPromiseDone (http://localhost:8080/resources/bower_components/bootstrap-tour/build/js/bootstrap-tour.js:703:19)
    at Tour.hideStep (http://localhost:8080/resources/bower_components/bootstrap-tour/build/js/bootstrap-tour.js:271:12)
    at Tour.goTo (http://localhost:8080/resources/bower_components/bootstrap-tour/build/js/bootstrap-tour.js:171:22)
```

Even refreshing doesn't make the error go away.

The offending line is this.backdrop.remove(); from:
```javascript
  Tour.prototype._hideBackground = function() {
      if (this.backdrop) {
        this.backdrop.remove();
        this.backdrop.overlay = null;
        return this.backdrop.backgroundShown = false;
      }
    };
```

Printing the backdrop object (this.backdrop) to console results in:
```
Object {overlay: null, $element: null, $background: null, backgroundShown: false, overlayElementShown: false}
```

Adding an additional check in the "if" for the remove function solves the problem apparently.

```javascript
  Tour.prototype._hideBackground = function() {
      if (this.backdrop && this.backdrop.remove) {
        this.backdrop.remove();
        this.backdrop.overlay = null;
        return this.backdrop.backgroundShown = false;
      }
    };
```

Is this fix correct or should I be concerned that other functionality might get broken?

**Comment by kblestarge** [comment]
I've been having the same problem.
Your fix worked for me. Thank you :)

**Comment by sorich87** [comment]
Fixed in f98650e129fe84dc2259fa2.
